crdt.go

```diff
@@ -311,22 +311,49 @@ func (store *Datastore) handleNext() {
 			continue
 		}
 
-		c, err := cid.Cast(data)
+		var bCastHeads []cid.Cid
+
+		// Make a list of heads we received
+		bcastData := pb.CRDTBroadcast{}
+		err = proto.Unmarshal(data, &bcastData)
 		if err != nil {
-			store.logger.Error(err)
-			continue
+			// Backwards compatibility
+			c, err2 := cid.Cast(data)
```

Comment:
Were these CIDv0 or CIDv1 before?

If they were CIDv1, this should be fine because field numbers must be >= 1 and the CID version, 0x1, decodes to field number 0, major type 1 (64bit thing).

If they were CIDv0, I don't think this will work. The first byte of a CIDv0 is 0x12 which decodes to field number 2 and field type 2 (length-delimited thing). Unfortunately, the _second_ byte is a varint length which is _also_ what the protobuf decoder would expect. That means the protobuf decoder will decode the CIDv0 as a valid protobuf and store it in `XXX_unrecognized`.

Reply:
I think CIDv0... hmm I will add a test and figure out the workaround.